packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -183,9 +184,26 @@ class _DayPeriodControl extends StatelessWidget {
 
   final _TimePickerFragmentContext fragmentContext;
 
-  void _handleChangeDayPeriod() {
+  void _togglePeriod() {
     final int newHour = (fragmentContext.selectedTime.hour + TimeOfDay.hoursPerPeriod) % TimeOfDay.hoursPerDay;
-    fragmentContext.onTimeChange(fragmentContext.selectedTime.replacing(hour: newHour));
+    final TimeOfDay newTime = fragmentContext.selectedTime.replacing(hour: newHour);
+    fragmentContext.onTimeChange(newTime);
+  }
+
+  void _setAm(BuildContext context) {
+    if (fragmentContext.selectedTime.period == DayPeriod.am) {
+      return;
+    }
+    _announceToAccessibility(context, MaterialLocalizations.of(context).anteMeridiemAbbreviation);
+    _togglePeriod();
+  }
+
+  void _setPm(BuildContext context) {
+    if (fragmentContext.selectedTime.period == DayPeriod.pm) {
+      return;
+    }
+    _announceToAccessibility(context, MaterialLocalizations.of(context).postMeridiemAbbreviation);
```

Comment:
Should we exclude these announcements on iOS since they are not necessary?

Reply:
Done.

Reply:
https://github.com/flutter/flutter/pull/13152/commits/00f7af9b122d09c7a47e49ab08557e083527189e